iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,320 @@
+
+    const auto status = m_senderRuntime->sendRequestToRouDi(m_sendBuffer, m_receiveBuffer);
+
+    EXPECT_EQ(true, status);
+}
+
+
+TEST_F(PoshRuntime_test, SendRequestToRouDi_InvalidMessage)
+{
+    m_sendBuffer << mqMessageTypeToString(MqMessageType::IMPL_INTERFACE) << std::string("/sender")
+                 << static_cast<uint32_t>(iox::capro::Interfaces::INTERNAL) << m_invalidRunnableName;
+
+    const auto status = m_senderRuntime->sendRequestToRouDi(m_sendBuffer, m_receiveBuffer);
+
+    EXPECT_EQ(false, status);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSender_ReturnValue)
+{
+    const auto senderPort = m_senderRuntime->getMiddlewareSender(
+        iox::capro::ServiceDescription(99u, 1u, 20u), m_runnableName, iox::runtime::PortConfigInfo(11u, 22u, 33u));
+
+    EXPECT_EQ(iox::capro::ServiceDescription(99u, 1u, 20u), senderPort->m_serviceDescription);
+    EXPECT_EQ(22u, senderPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(33u, senderPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSender_DefaultArgs)
+{
+    const auto senderPort = m_senderRuntime->getMiddlewareSender(iox::capro::ServiceDescription(99u, 1u, 20u));
+
+    EXPECT_EQ(0u, senderPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(0u, senderPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSender_SenderlistOverflow)
+{
+    m_errorHandlerCalled = false;
+
+    for (uint32_t i = 0u; i < iox::MAX_PORT_NUMBER; ++i)
+    {
+        m_senderRuntime->getMiddlewareSender(iox::capro::ServiceDescription(i, i + 1u, i + 2u));
+    }
+
+    EXPECT_TRUE(m_errorHandlerCalled);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareReceiver_ReturnValue)
+{
+    auto receiverPort = m_receiverRuntime->getMiddlewareReceiver(
+        iox::capro::ServiceDescription(99u, 1u, 20u), m_runnableName, iox::runtime::PortConfigInfo(11u, 22u, 33u));
+
+    EXPECT_EQ(iox::capro::ServiceDescription(99u, 1u, 20u), receiverPort->m_serviceDescription);
+    EXPECT_EQ(22u, receiverPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(33u, receiverPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareReceiver_DefaultArgs)
+{
+    auto receiverPort = m_receiverRuntime->getMiddlewareReceiver(iox::capro::ServiceDescription(99u, 1u, 20u));
+
+    EXPECT_EQ(0u, receiverPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(0u, receiverPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareReceiver_ReceiverlistOverflow)
+{
+    m_errorHandlerCalled = false;
+
+    for (uint32_t i = 0u; i < iox::MAX_PORT_NUMBER + 1; ++i)
+    {
+        m_senderRuntime->getMiddlewareReceiver(iox::capro::ServiceDescription(i, i + 1u, i + 2u));
+    }
+
+    EXPECT_TRUE(m_errorHandlerCalled);
+}
+
+
+TEST_F(PoshRuntime_test, GetServiceRegistryChangeCounter_ReturnValue)
+{
+    auto runnableData = m_senderRuntime->getServiceRegistryChangeCounter();
+
+    // Roudi internally  calls 5 servieces before application registers its service
```

Comment:
This test is probably brittle and will break when internals change as it relies on implementation details. As far as I know, the only requirement for the counter is to monotonically change when services register/deregister, i.e. some registration status changes. Personally I think only this should be tested: whether the counter changes (by increasing) if something happens that is supposed to cause a change (as mentioned above).

Reply:
I see this is done below, in this case I think that this test can be removed entirely, we do not care for the initial value to be 5.